File: ui/trading-app/src/types/CypherBQuote.ts
import { MfiResult } from "./Mfi";
import { Quote } from "./Quote";
import { SrsiSignal } from "./SrsiSignal";
import { WaveTrendSignal } from "./WaveTrendSignal";

export interface CypherBQuote {
  ohlc: Quote;
  waveTrendSignal: WaveTrendSignal;
  mfiResult: MfiResult;
  srsiSignal: SrsiSignal;
}

File: src/TradingApp.Module.Quotes/Application/Models/CypherBQuote.cs
using TradingApp.Module.Quotes.Domain.Enums;

namespace TradingApp.Module.Quotes.Application.Models;
using TradingApp.Module.Quotes.Contract.Models;

public record CypherBQuote(Quote Ohlc, WaveTrendSignal WaveTrendSignal, MfiResult MfiResult, SrsiSignal SrsiSignal);

public record WaveTrendSignal(decimal Wt1, decimal Wt2, decimal? Vwap, TradeAction TradeAction);


public record SrsiSignal(decimal StochK, decimal StochD, TradeAction TradeAction);